osscan2.cc

```diff
@@ -3411,7 +3411,7 @@ bool HostOsScan::get_tcpopt_string(struct tcp_hdr *tcp, int mss, char *result, i
       memcpy(&tmpshort, q, 2);
       /*  if (ntohs(tmpshort) == mss) */
       /*    *p++ = 'E'; */
-      sprintf(p, "%hX", ntohs(tmpshort));
+      sprintf(p, "%hhX", ntohs(tmpshort));
```

Comment:
Indeed you are right. I made the modification a bit too fast, it was actually a few lines after that the warning was occurring. Thank you for the comment!